services/core/businessconfig/src/main/java/org/lfenergy/operatorfabric/businessconfig/services/ProcessesService.java

```diff
@@ -202,11 +203,17 @@ public Resource fetchResource(String processId, ResourceTypeEnum type, String ve
 
         if (process == null)
             throw new FileNotFoundException("Unknown version (" + finalVersion + ") for " + processId);
-        validateResourceParameters(processId, type, name, finalVersion, locale);
+
+        if (type.isLocalized() && locale == null)
+            throw new FileNotFoundException("Unable to determine resource for undefined locale");
+
         String finalName;
         if (type == ResourceTypeEnum.I18N) {
             finalName = locale;
         } else {
+            if (!SecurityUtils.checkFilename(name)) {
+                throw new FileNotFoundException("Filename contains forbidden characters");
+            }
```

Comment:
You shall test locale and version parameters to not contains forbidden characters

Reply:
Ok to check the locale. I think that the version parameter is already validated against the available process versions and if it does not match an Exception is thrown.